espresso-server/app/src/test/java/io/appium/espressoserver/test/helpers/w3c/PointerDispatchTest.java

```diff
@@ -46,24 +47,26 @@
     private PointerInputState pointerInputSource;
 
     @Test
-    public void shouldNoopPointerMoveIfNoButtons() throws ExecutionException, InterruptedException {
+    public void shouldNoopPointerMoveIfNoButtons() throws ExecutionException, InterruptedException, AppiumException {
         DummyW3CActionAdapter dummyW3CActionAdapter = new DummyW3CActionAdapter();
         pointerInputSource = new PointerInputState();
         pointerInputSource.setType(PointerType.TOUCH);
 
         ExecutorService executorService = Executors.newSingleThreadExecutor();
-        Callable<Void> callable = performPointerMove(
+        Callable<BaseDispatchResult> callable = performPointerMove(
             dummyW3CActionAdapter, "any", pointerInputSource,
             100, 10, 20, 30, 40, System.currentTimeMillis(),
             new KeyInputState()
         );
-        executorService.submit(callable).get();
+        BaseDispatchResult dispatchResult = executorService.submit(callable).get();
+        dispatchResult.perform();
         executorService.shutdown();
         assertEquals(dummyW3CActionAdapter.getPointerMoveEvents().size(), 1);
     }
 
     @Test
-    public void shouldDoOneMoveIfDurationZero() throws ExecutionException, InterruptedException {
+    public void shouldDoOneMoveIfDurationZero()
+            throws ExecutionException, InterruptedException, AppiumException {
```

Comment:
shouldn't ExecutionException/InterruptedException be covered into  AppiumException?

Reply:
`ExecutionException` and `InterruptedException` are thrown by `executorService.submit(callable).get()`

Reply:
I know, but we can transform them in try-catch

Reply:
Does that matter for a unit test though?

Reply:
for testing apparently not

Reply:
I'm still trying to understand the finer points of Java...

I understand that in that we should be catching exceptions like InterruptedException and ExecutionException and throw them as AppiumException because the exceptions need to translate to proper webdriver errors.

In this case, though, it's just a unit test that doesn't get called by other methods, so does it make a difference what exceptions are thrown?